### PR TITLE
feat: [CDS-76595]: Change code pulse annotations where ever needed

### PR DESCRIPTION
**Pull request by abhinavmittal-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51475)

### CODE REVIEW

File: 440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/servicenow/ServiceNowRefreshTokenAuthentication.java [hunk 1]
```diff
@@ -20,7 +20,8 @@
 import lombok.experimental.FieldNameConstants;
 import org.springframework.data.annotation.TypeAlias;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
+@CodePulse(
```
Comment (puthrayaharness) [inline 440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/servicenow/ServiceNowRefreshTokenAuthentication.java:23]: This should be approvals

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java [hunk 2]
```diff
@@ -28,7 +28,8 @@
 import lombok.Value;
 import lombok.experimental.FieldDefaults;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = false,
```
Comment (puthrayaharness) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java:31]: This falls under Approvals

File: 960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenAuthNgHelper.java [hunk 3]
```diff
@@ -57,7 +57,8 @@
  * Tested using ServiceNow and Okta as authentication severs.
  */
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true,
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenAuthNgHelper.java:60]: This falls under Approvals only

File: 960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenResponseHandler.java [hunk 4]
```diff
@@ -23,7 +23,8 @@
 import lombok.extern.slf4j.Slf4j;
 import retrofit2.Response;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
+@CodePulse(
```
Comment (puthrayaharness) [inline 960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenResponseHandler.java:26]: Approvals only

File: 980-commons/src/main/java/io/harness/expression/ImageSecretFunctor.java [hunk 5]
```diff
@@ -19,7 +19,8 @@
 
 import java.util.concurrent.Future;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
```
Comment (puthrayaharness) [inline 980-commons/src/main/java/io/harness/expression/ImageSecretFunctor.java:22]: This falls under Artifacts

File: 999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java [hunk 6]
```diff
@@ -32,5 +32,7 @@ public enum HarnessModuleComponent {
   CDS_TEMPLATE_LIBRARY,
   CDS_TRADITIONAL,
   CDS_TRIGGERS,
-  CDS_SLACK
+  CDS_SLACK,
```
Comment (puthrayaharness) [inline 999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java:35]: Please remove Slack. Slack should only be an Approvals component. The component should match Jira component.
Please remove TO_BE_ADDED

File: pipeline-service/service/src/main/java/io/harness/pms/jira/JiraStepHelperServiceImpl.java [hunk 7]
```diff
@@ -53,7 +53,8 @@
 import java.util.List;
 import java.util.Optional;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true,
```
Comment (puthrayaharness) [inline pipeline-service/service/src/main/java/io/harness/pms/jira/JiraStepHelperServiceImpl.java:56]: Remove common steps. It is approvals only

File: pipeline-service/service/src/main/java/io/harness/pms/servicenow/ServiceNowStepHelperServiceImpl.java [hunk 8]
```diff
@@ -64,7 +64,8 @@
 import java.util.Map;
 import java.util.Optional;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
+@CodePulse(
```
Comment (puthrayaharness) [inline pipeline-service/service/src/main/java/io/harness/pms/servicenow/ServiceNowStepHelperServiceImpl.java:67]: Approvals

**Review by puthrayaharness — CHANGES_REQUESTED** [review]

File: 960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenConstants.java [hunk 9]
```diff
@@ -10,7 +10,8 @@
 import io.harness.annotations.dev.HarnessModuleComponent;
 import io.harness.annotations.dev.ProductModule;
 
-@CodePulse(module = ProductModule.CDS, unitCoverageRequired = true, components = {HarnessModuleComponent.CDS_APPROVALS})
+@CodePulse(module = ProductModule.CDS, unitCoverageRequired = false,
+    components = {HarnessModuleComponent.CDS_APPROVALS, HarnessModuleComponent.CDS_COMMON_STEPS})
```
Comment (naman-goenka) [inline 960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenConstants.java:14]: This is currently used in approvals only

**Review by abhinavmittal-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51475)* status: 40 file(s) to review. 1 change requests. 9. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/servicenow/ServiceNowRefreshTokenAuthentication.java` line 23 at r1](https://reviewable.io/reviews/harness/harness-core/51475#gh-1295434892:-NcLjh_RCkkgJQiglz4W:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd3381c906a24ac5e6c4b5287ea60519a147379e/440-connector-nextgen/src/main/java/io/harness/connector/entities/embedded/servicenow/ServiceNowRefreshTokenAuthentication.java#L23)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

This should be approvals
</blockquote></details>

Done.
___
*[`950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java` line 31 at r1](https://reviewable.io/reviews/harness/harness-core/51475#gh-1295454184:-NcLjpBP2tSx-LM5OD7W:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd3381c906a24ac5e6c4b5287ea60519a147379e/950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/jira/JiraTaskNGParameters.java#L31)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

This falls under Approvals
</blockquote></details>

Done.
___
*[`960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenAuthNgHelper.java` line 60 at r1](https://reviewable.io/reviews/harness/harness-core/51475#gh-1295454824:-NcLjqgW6cjdNf2Ss729:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd3381c906a24ac5e6c4b5287ea60519a147379e/960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenAuthNgHelper.java#L60)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

This falls under Approvals only
</blockquote></details>

Done.
___
*[`960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenConstants.java` line 14 at r1](https://reviewable.io/reviews/harness/harness-core/51475#gh-1296093862:-NcLjrWt3aATahp_ACOM:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd3381c906a24ac5e6c4b5287ea60519a147379e/960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenConstants.java#L14)):*
<details><summary><i>Previously, naman-goenka (Naman Goenka) wrote…</i></summary><blockquote>

This is currently used in approvals only
</blockquote></details>

Done.
___
*[`980-commons/src/main/java/io/harness/expression/ImageSecretFunctor.java` line 0 at r1](https://reviewable.io/reviews/harness/harness-core/51475#gh-1295455843:-NcLjzT10Y1BX0djC6nU:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd3381c906a24ac5e6c4b5287ea60519a147379e/980-commons/src/main/java/io/harness/expression/ImageSecretFunctor.java#L0)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

This falls under Artifacts
</blockquote></details>

Done.
___
*[`999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java` line 35 at r1](https://reviewable.io/reviews/harness/harness-core/51475#gh-1295456827:-NcLk0706jXuafngfDsm:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd3381c906a24ac5e6c4b5287ea60519a147379e/999-annotations/src/main/java/io/harness/annotations/dev/HarnessModuleComponent.java#L35)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

Please remove Slack. Slack should only be an Approvals component. The component should match Jira component.
Please remove TO_BE_ADDED
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/jira/JiraStepHelperServiceImpl.java` line 56 at r1](https://reviewable.io/reviews/harness/harness-core/51475#gh-1295457169:-NcLk1j945bnpSzLx2tV:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd3381c906a24ac5e6c4b5287ea60519a147379e/pipeline-service/service/src/main/java/io/harness/pms/jira/JiraStepHelperServiceImpl.java#L56)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

Remove common steps. It is approvals only
</blockquote></details>

Done.
___
*[`960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenResponseHandler.java` line 26 at r1](https://reviewable.io/reviews/harness/harness-core/51475#gh-1295455142:-NcLjx0zCb1ziZTxvhiH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd3381c906a24ac5e6c4b5287ea60519a147379e/960-api-services/src/main/java/io/harness/servicenow/auth/refreshtoken/RefreshTokenResponseHandler.java#L26)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

Approvals only
</blockquote></details>

Done.
___
*[`pipeline-service/service/src/main/java/io/harness/pms/servicenow/ServiceNowStepHelperServiceImpl.java` line 67 at r1](https://reviewable.io/reviews/harness/harness-core/51475#gh-1295457321:-NcLk9ocCbOqy9jXY9QH:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/dd3381c906a24ac5e6c4b5287ea60519a147379e/pipeline-service/service/src/main/java/io/harness/pms/servicenow/ServiceNowStepHelperServiceImpl.java#L67)):*
<details><summary><i>Previously, puthrayaharness (Deepak Puthraya) wrote…</i></summary><blockquote>

Approvals
</blockquote></details>

Done.

**Review by puthrayaharness — APPROVED** [review]

**Comment by puthrayaharness** [comment]
trigger uts

**Review by vcalasansh — APPROVED** [review]
> Reviewed 16 of 39 files at r1, 24 of 24 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51475)* status: 40 file(s) to review. 9. unresolved discussions.

**Comment by abhinavmittal-harness** [comment]
trigger messagecheck

**Comment by abhinavmittal-harness** [comment]
trigger messagecheck

**Pull request merged**